### PR TITLE
feat: tracking notifications

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Fixed the mismatch of using source label for community picks modal as new source.
- Events listed in the taxonomy sheet.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/components/layout/MainLayoutHeader.tsx [hunk 1]
```diff
@@ -101,6 +111,7 @@ function MainLayoutHeader({
           {!hideButton && user && (
             <LinkWithTooltip
               tooltip={{ placement: 'left', content: 'Notifications' }}
+              onClick={onNavigateNotifications}
```
Comment (idoshamun) [inline packages/shared/src/components/layout/MainLayoutHeader.tsx:114]: Shouldn't we add the same callback to the mobile button as well?

File: packages/shared/src/components/notifications/EnableNotification.tsx [hunk 2]
```diff
@@ -7,21 +7,23 @@ import CloseButton from '../CloseButton';
...
+  NotificationsPage = 'notifications page',
+  NewComment = 'new comment',
+  CommunityPicks = 'community picks modal',
 }
 
 const DISMISS_BROWSER_PERMISSION = 'dismissBrowserPermission';
 
 type DismissBrowserPermissions = Record<NotificationPromptSource, boolean>;
 
 const DEFAULT_CACHE_VALUE: DismissBrowserPermissions = {
```
Comment (idoshamun) [inline packages/shared/src/components/notifications/EnableNotification.tsx:23]: A general comment that you should address separately, dismissing is not per prompt but cross app.
Meaning if I dismissed the widget I should never see it again.

Comment (sshanzel) [reply packages/shared/src/components/notifications/EnableNotification.tsx:23]: Cool. Let me refactor this one.

File: packages/shared/src/components/layout/MainLayoutHeader.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/layout/MainLayoutHeader.tsx:114]: Yes, it should be the line below, from `FooterNavBar`:
https://github.com/dailydotdev/apps/blob/a50edc87ec402b47b60a2ac8f51efdd044cf7669/packages/webapp/components/FooterNavBar.tsx#L93-L103

File: packages/webapp/components/layouts/AccountLayout/SidebarNav.tsx [hunk 3]
```diff
@@ -50,6 +61,15 @@ function SidebarNav({
     };
   }, [router]);
 
+  const onItemClick: Partial<FunctionRecord<AccountPage>> = {
```
Comment (idoshamun) [inline packages/webapp/components/layouts/AccountLayout/SidebarNav.tsx:64]: Not sure we should track it here. This leads to the notification settings if I understand correctly

File: packages/webapp/pages/notifications.tsx [hunk 4]
```diff
@@ -50,12 +56,31 @@ const Notifications = (): ReactElement => {
...
+      extra: JSON.stringify({ origin: Origin.NonRealTime, type }),
+    });
+  };
+
+  useEffect(() => {
+    trackEvent({
+      event_name: AnalyticsEvent.Impression,
+      extra: JSON.stringify({ notifications_number: unreadCount }),
+    });
+  }, []);
```
Comment (idoshamun) [inline packages/webapp/pages/notifications.tsx:82]: This should be tracked whenever we show the bell icon with at least one unread count, not here

**Review by idoshamun — CHANGES_REQUESTED** [review]
Let's add also the notification id to the click notification event. We can utilize the `target_id` field for that

File: packages/webapp/components/layouts/AccountLayout/SidebarNav.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/SidebarNav.tsx:64]: Ah, correct. This was a mistake. Let me fix this quickly.

Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/SidebarNav.tsx:64]: Ah, correct. This was a mistake. Let me fix this quickly. I saw the possible values to have `Sidebar` so I mistakenly had added this too.

File: packages/webapp/pages/notifications.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/webapp/pages/notifications.tsx:82]: Oh, cool. I misunderstood it for if the icon is filled which only happens here. Thanks for clearing it up.

File: packages/shared/src/contexts/NotificationsContext.tsx [hunk 5]
```diff
@@ -144,6 +150,18 @@ export const NotificationsContextProvider = ({
     });
   }, [isInitialized, isInitializing, user]);
 
+  useEffect(() => {
+    if (!firstLoad || unreadCount === 0 || hasTrackedImpression) {
+      return;
+    }
+
+    trackEvent({
```
Comment (idoshamun) [inline packages/shared/src/contexts/NotificationsContext.tsx:158]: You're missing a `target_type: 'notifications icon'`

**Review by idoshamun — APPROVED** [review]
One quick fix and we can merge!

**Pull request merged**